lib/kubernetes-deploy/kubernetes_resource/service.rb

```diff
@@ -16,7 +16,11 @@ def sync
     end
 
     def deploy_succeeded?
-      @num_endpoints > 0
+      if related_deployment_replicas
```

Comment:
The purpose of this change is to be able to handle zero-replica deployments (that have been temporarily scaled down for some reason). I'm second-guessing the what I've done here though--I think it might be too aggressive. If the deployment has rolled out and the service has _some_ endpoints, it's not really valuable to hold up the deploy, is it?

WDTY think about this logic instead:

```ruby
if exposes_zero_replica_deployment?
 @num_endpoints == 0
else
  @num_endpoints > 0
end
```